libevmasm/Assembly.cpp

```diff
@@ -74,6 +76,209 @@ unsigned Assembly::codeSize(unsigned subTagSize) const
 	}
 }
 
+void Assembly::importAssemblyItemsFromJSON(Json::Value const& _code)
+{
+	solAssert(m_items.empty());
+	solRequire(_code.isArray(), AssemblyImportException, "Supplied JSON is not an array.");
+	for (auto current = begin(_code); current != end(_code); ++current)
+	{
+		auto const& item = m_items.emplace_back(createAssemblyItemFromJSON(*current, m_sourceList));
+		if (item == Instruction::JUMPDEST)
+			solThrow(AssemblyImportException, "JUMPDEST instruction without a tag");
+		else if (item.type() == AssemblyItemType::Tag)
+			solRequire(
+				++current != end(_code) &&
+					createAssemblyItemFromJSON(*current, m_sourceList) == Instruction::JUMPDEST,
+				AssemblyImportException,
+				"JUMPDEST expected after tag."
+			);
+	}
+}
+
+AssemblyItem Assembly::createAssemblyItemFromJSON(Json::Value const& _json, std::vector<std::string> const& _sourceList)
+{
+	solRequire(_json.isObject(), AssemblyImportException, "Supplied JSON is not an object.");
+	static set<string> const validMembers{"name", "begin", "end", "source", "value", "modifierDepth", "jumpType"};
+	for (auto const& member: _json.getMemberNames())
+		solRequire(
+			validMembers.count(member),
+			AssemblyImportException,
+			"Unknown member '" + member + "'. Valid members are " +
+			solidity::util::joinHumanReadable(validMembers, ", ") + "."
+		);
+	solRequire(isOfType<string>(_json["name"]), AssemblyImportException, "Member 'name' not of type string.");
+	solRequire(isOfTypeIfExists<int>(_json, "begin"), AssemblyImportException, "Optional member 'begin' not of type int.");
+	solRequire(isOfTypeIfExists<int>(_json, "end"), AssemblyImportException, "Optional member 'end' not of type int.");
+	solRequire(isOfTypeIfExists<int>(_json, "source"), AssemblyImportException, "Optional member 'source' not of type int.");
+	solRequire(isOfTypeIfExists<string>(_json, "value"), AssemblyImportException, "Optional member 'value' not of type string.");
+	solRequire(
+		isOfTypeIfExists<int>(_json, "modifierDepth"),
+		AssemblyImportException,
+		"Optional member 'modifierDepth' not of type int."
+	);
+	solRequire(
+		isOfTypeIfExists<string>(_json, "jumpType"),
+		AssemblyImportException,
+		"Optional member 'jumpType' not of type string."
+	);
+
+	string name = getOrDefault<string>(_json["name"], "");
```

Comment:
Isn't `name` mandatory? Won't the validation above reject it already?
```suggestion
	string name = get<string>(_json["name"]);
```